diff --git a/apps/projectGenerator b/apps/projectGenerator
--- a/apps/projectGenerator
+++ b/apps/projectGenerator
@@ -1 +1 @@
-Subproject commit 74afcd48555ba9b3d0c430b2853ddad306a89b63
+Subproject commit 74afcd48555ba9b3d0c430b2853ddad306a89b63-dirty
diff --git a/libs/openFrameworks/communication/ofSerial.cpp b/libs/openFrameworks/communication/ofSerial.cpp
index 67b85bec3..d6cef78ea 100644
--- a/libs/openFrameworks/communication/ofSerial.cpp
+++ b/libs/openFrameworks/communication/ofSerial.cpp
@@ -18,7 +18,9 @@
 #endif
 
 #ifdef TARGET_LINUX
-	#include <linux/serial.h>
+#include <sys/termios.h>
+#include <sys/ioctl.h>
+	/*#include <linux/serial.h>*/
 #endif
 
 // FIXME: check if this i sexclusive to windows and move to the right target if yes.
@@ -400,12 +402,14 @@ bool ofSerial::setup(string portName, int baud){
 		#endif
 		tcsetattr(fd, TCSANOW, &options);
 		#ifdef TARGET_LINUX
+		#ifndef __FreeBSD__
 			struct serial_struct kernel_serial_settings;
 			if (ioctl(fd, TIOCGSERIAL, &kernel_serial_settings) == 0) {
 				kernel_serial_settings.flags |= ASYNC_LOW_LATENCY;
 				ioctl(fd, TIOCSSERIAL, &kernel_serial_settings);
 			}
 		#endif
+		#endif
 		bInited = true;
 		ofLogNotice("ofSerial") << "opened " << portName << " sucessfully @ " << baud << " bps";
 
diff --git a/libs/openFrameworks/sound/ofOpenALSoundPlayer.cpp b/libs/openFrameworks/sound/ofOpenALSoundPlayer.cpp
index e14ebb63d..63a5f1406 100644
--- a/libs/openFrameworks/sound/ofOpenALSoundPlayer.cpp
+++ b/libs/openFrameworks/sound/ofOpenALSoundPlayer.cpp
@@ -735,8 +735,10 @@ void ofOpenALSoundPlayer::unload(){
 		std::unique_lock<std::mutex> lock(mutex);
 
 		// Delete sources before buffers.
-		alDeleteSources(sources.size(),&sources[0]);
-		alDeleteBuffers(buffers.size(),&buffers[0]);
+		if(!sources.empty())
+			alDeleteSources(sources.size(),&sources[0]);
+		if(!buffers.empty())
+			alDeleteBuffers(buffers.size(),&buffers[0]);
 
 		sources.clear();
 		buffers.clear();
diff --git a/libs/openFrameworks/utils/ofFileUtils.cpp b/libs/openFrameworks/utils/ofFileUtils.cpp
index f699f7830..b6c6642c1 100644
--- a/libs/openFrameworks/utils/ofFileUtils.cpp
+++ b/libs/openFrameworks/utils/ofFileUtils.cpp
@@ -12,6 +12,8 @@
 	#include <mach-o/dyld.h>       /* _NSGetExecutablePath */
 	#include <limits.h>        /* PATH_MAX */
 #endif
+#include <limits.h>
+#include <sys/sysctl.h>
 
 namespace fs = of::filesystem;
 
@@ -1872,6 +1874,7 @@ std::string ofFilePath::join(const fs::path& path1, const fs::path& path2){
 fs::path ofFilePath::getCurrentExePathFS(){
 	#if defined(TARGET_LINUX) || defined(TARGET_ANDROID)
 		char buff[FILENAME_MAX];
+		#ifndef __FreeBSD__
 		ssize_t size = readlink("/proc/self/exe", buff, sizeof(buff) - 1);
 		if (size == -1){
 			ofLogError("ofFilePath") << "getCurrentExePath(): readlink failed with error " << errno;
@@ -1880,6 +1883,13 @@ fs::path ofFilePath::getCurrentExePathFS(){
 			buff[size] = '\0';
 			return buff;
 		}
+	#else
+	static int pathname[] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
+	pathname[3] = getpid();
+	
+	size_t bsize = FILENAME_MAX;
+        sysctl( pathname, sizeof(pathname)/sizeof(pathname[0]), buff,&bsize , NULL, 0 );
+	#endif
 	#elif defined(TARGET_OSX)
 		char path[FILENAME_MAX];
 		uint32_t size = sizeof(path);
diff --git a/libs/openFrameworks/utils/ofUtils.cpp b/libs/openFrameworks/utils/ofUtils.cpp
index c362239b6..22be36ec3 100644
--- a/libs/openFrameworks/utils/ofUtils.cpp
+++ b/libs/openFrameworks/utils/ofUtils.cpp
@@ -1093,7 +1093,8 @@ string ofSystem(const string & command) {
 ofTargetPlatform ofGetTargetPlatform() {
 #ifdef TARGET_LINUX
 	string arch = ofSystem("uname -m");
-	if (ofIsStringInString(arch, "x86_64")) {
+	fprintf(stderr,"%s:%d %s\n",__FILE__,__LINE__,arch.c_str());
+	if (ofIsStringInString(arch, "amd64")) {
 		return OF_TARGET_LINUX64;
 	} else if (ofIsStringInString(arch, "armv6l")) {
 		return OF_TARGET_LINUXARMV6L;
diff --git a/libs/openFrameworks/video/ofGstVideoGrabber.cpp b/libs/openFrameworks/video/ofGstVideoGrabber.cpp
index 2ed999464..5b7c3eb05 100644
--- a/libs/openFrameworks/video/ofGstVideoGrabber.cpp
+++ b/libs/openFrameworks/video/ofGstVideoGrabber.cpp
@@ -33,34 +33,12 @@ extern "C" {
 #include <fcntl.h>
 #include <unistd.h>
 #include <sys/ioctl.h>
-
-#include <linux/version.h>
-
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,35)
-	#include <linux/videodev2.h>
-	#define VIDIOCGCAP              _IOR('v',1,struct video_capability)     /* Get capabilities */
-
-	struct video_capability
-	{
-			 char name[32];
-			 int type;
-			 int channels;   /* Num channels */
-			 int audios;     /* Num audio devices */
-			 int maxwidth;   /* Supported width */
-			 int maxheight;  /* And height */
-			 int minwidth;   /* Supported width */
-			 int minheight;  /* And height */
-	};
-#else
-	#include <linux/videodev.h>
-#endif
-
-
-#endif
+#include <linux/videodev2.h>
+#include <linux/videodev.h>
 
 using std::string;
 using std::vector;
-
+#endif
 static void get_video_devices (ofGstCamData & cam_data)
 {
 #ifdef TARGET_LINUX
diff --git a/libs/openFrameworksCompiled/project/makefileCommon/compile.core.mk b/libs/openFrameworksCompiled/project/makefileCommon/compile.core.mk
index 78a80027e..72bc2f737 100644
--- a/libs/openFrameworksCompiled/project/makefileCommon/compile.core.mk
+++ b/libs/openFrameworksCompiled/project/makefileCommon/compile.core.mk
@@ -68,7 +68,7 @@ ALL_CFLAGS += $(OF_CORE_BASE_CFLAGS)
 # add the defines
 ALL_CFLAGS += $(OF_CORE_DEFINES_CFLAGS)
 # add the include cflags
-ALL_CFLAGS += $(OF_CORE_INCLUDES_CFLAGS)
+ALL_CFLAGS += $(OF_CORE_INCLUDES_CFLAGS) -I./../../../scripts/apothecary/out/utf8/include
 # clean up all extra whitespaces in the CFLAGS
 CFLAGS += $(strip $(ALL_CFLAGS))
 CXXFLAGS += $(OF_CORE_BASE_CXXFLAGS)
diff --git a/libs/openFrameworksCompiled/project/makefileCommon/compile.project.mk b/libs/openFrameworksCompiled/project/makefileCommon/compile.project.mk
index 95af9d9d3..eb7aa6bda 100644
--- a/libs/openFrameworksCompiled/project/makefileCommon/compile.project.mk
+++ b/libs/openFrameworksCompiled/project/makefileCommon/compile.project.mk
@@ -1,5 +1,4 @@
 .DEFAULT_GOAL=Release
-
 # define the OF_SHARED_MAKEFILES location
 OF_SHARED_MAKEFILES_PATH=$(OF_ROOT)/libs/openFrameworksCompiled/project/makefileCommon
 
@@ -9,7 +8,7 @@ ifndef APPNAME
 endif
 
 include $(OF_SHARED_MAKEFILES_PATH)/config.shared.mk
-
+CFLAGS+=-I$(OF_ROOT)/scripts/apothecary/out/utf8/include/ -g -O0
 # Name TARGET
 ifeq ($(findstring Debug,$(MAKECMDGOALS)),Debug)
 	TARGET_NAME = Debug
diff --git a/libs/openFrameworksCompiled/project/makefileCommon/config.linux.common.mk b/libs/openFrameworksCompiled/project/makefileCommon/config.linux.common.mk
index c31b10bd3..95ba21c4e 100755
--- a/libs/openFrameworksCompiled/project/makefileCommon/config.linux.common.mk
+++ b/libs/openFrameworksCompiled/project/makefileCommon/config.linux.common.mk
@@ -214,7 +214,7 @@ endif
 
 ifndef PROJECT_OPTIMIZATION_CFLAGS_RELEASE
 	# RELEASE Debugging options (http://gcc.gnu.org/onlinedocs/gcc/Debugging-Options.html)
-	PLATFORM_OPTIMIZATION_CFLAGS_RELEASE = -O3
+	PLATFORM_OPTIMIZATION_CFLAGS_RELEASE = -O0 -g
 
 	#ifneq ($(LINUX_ARM),1)
 	#	PLATFORM_OPTIMIZATION_CFLAGS_RELEASE += -march=native -mtune=native
@@ -355,6 +355,8 @@ PLATFORM_PKG_CONFIG_LIBRARIES += freetype2
 PLATFORM_PKG_CONFIG_LIBRARIES += fontconfig
 PLATFORM_PKG_CONFIG_LIBRARIES += sndfile
 PLATFORM_PKG_CONFIG_LIBRARIES += openal
+PLATFORM_PKG_CONFIG_LIBRARIES += tess2
+PLATFORM_PKG_CONFIG_LIBRARIES += kissfft-float
 
 
 ifeq "$(shell pkg-config --exists openssl && echo 1)" "1"
diff --git a/libs/openFrameworksCompiled/project/makefileCommon/config.shared.mk b/libs/openFrameworksCompiled/project/makefileCommon/config.shared.mk
index 4ab8df1f0..585e9c213 100644
--- a/libs/openFrameworksCompiled/project/makefileCommon/config.shared.mk
+++ b/libs/openFrameworksCompiled/project/makefileCommon/config.shared.mk
@@ -21,14 +21,15 @@
 SHELL ?= /bin/sh
 OF_ROOT ?= ../../..
 PLATFORM_VARIANT ?= default
-
 # ifeq ($(CC),$(EMSCRIPTEN)/emcc)
 ifeq ($(findstring emcc, $(CC)),emcc)
 	PLATFORM_OS=emscripten
 endif
+PLATFORM_OS= linux
 PLATFORM_OS ?= $(shell uname -s)
 
 HOST_OS=$(shell uname -s)
+HOST_OS=linux
 
 ifdef MAKEFILE_DEBUG
     $(info HOST_OS=${HOST_OS})
@@ -82,8 +83,8 @@ endif
 # if not defined, construct the default PLATFORM_LIB_SUBPATH
 ifndef PLATFORM_LIB_SUBPATH
 	# determine from the arch
-	ifeq ($(PLATFORM_OS),Linux)
-		ifeq ($(PLATFORM_ARCH),x86_64)
+	ifeq ($(PLATFORM_OS),linux)
+		ifeq ($(PLATFORM_ARCH),amd64)
 			PLATFORM_LIB_SUBPATH=linux64
 		else ifeq ($(PLATFORM_ARCH),armv6l)
 			PLATFORM_LIB_SUBPATH=linuxarmv6l
diff --git a/scripts/apothecary b/scripts/apothecary
--- a/scripts/apothecary
+++ b/scripts/apothecary
@@ -1 +1 @@
-Subproject commit fae96e615875a701d48dc29cf4a3be24bc472a07
+Subproject commit fae96e615875a701d48dc29cf4a3be24bc472a07-dirty
diff --git a/scripts/dev/download_libs.sh b/scripts/dev/download_libs.sh
index d0508f494..b2c3b5cf8 100755
--- a/scripts/dev/download_libs.sh
+++ b/scripts/dev/download_libs.sh
@@ -139,13 +139,13 @@ while [[ $# -gt 0 ]]; do
     esac
     shift # past argument or value
 done
-
+PLATFORM=linux
 if [[ "$TARGET" != "" ]] && [[ "$PLATFORM" == "" ]]; then
     PLATFORM=$TARGET
 fi
 
 if [ "$PLATFORM" == "" ]; then
-    OS=$(uname)
+    OS="Linux"
     if [ "$OS" == "Linux" ]; then
         PLATFORM="linux"
     elif [ "$OS" == "Darwin" ]; then
diff --git a/scripts/linux/buildAllExamples.sh b/scripts/linux/buildAllExamples.sh
index 7048eefa7..1d5971595 100755
--- a/scripts/linux/buildAllExamples.sh
+++ b/scripts/linux/buildAllExamples.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/local/bin/bash
 
 export LC_ALL=C
 
@@ -18,14 +18,14 @@ do
         echo -----------------------------------------------------------------
         echo building  $example
 
-        #projectGenerator .
-        make Debug -j -C "$example"
+        projectGenerator  -o $example
+        gmake Debug -j -C "$example"
         ret=$?
         if [ $ret -ne 0 ]; then
             echo error compiling $example
             exit
         fi
-        make Release -j -C "$example"
+        gmake Release -j -C "$example"
         ret=$?
         if [ $ret -ne 0 ]; then
             echo error compiling $example
diff --git a/scripts/linux/cleanAllExamples.sh b/scripts/linux/cleanAllExamples.sh
index 11f1ee922..94dc67a8e 100755
--- a/scripts/linux/cleanAllExamples.sh
+++ b/scripts/linux/cleanAllExamples.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/local/bin/bash
 
 for category in $( find ../../examples/ -maxdepth 1 -type d )
 do
diff --git a/scripts/linux/compileOF.sh b/scripts/linux/compileOF.sh
index 3f37c9974..473fc95fb 100755
--- a/scripts/linux/compileOF.sh
+++ b/scripts/linux/compileOF.sh
@@ -1,9 +1,10 @@
-#!/usr/bin/env bash
+#!/usr/local/bin/bash
+set -x
 
 export LC_ALL=C
 
 ARCH=$(uname -m)
-if [ "$ARCH" = "x86_64" ]; then
+if [ "$ARCH" = "amd64" ]; then
         LIBSPATH=linux64
 else
         LIBSPATH=linux
@@ -25,7 +26,7 @@ while getopts tj: opt ; do
 done
 
 cd "${SCRIPTPATH}/../../libs/openFrameworksCompiled/project"
-make -j$JOBS Debug
+gmake -j$JOBS Debug
 exit_code=$?
 if [ $exit_code != 0 ]; then
   echo "there has been a problem compiling Debug OF library"
@@ -34,7 +35,7 @@ if [ $exit_code != 0 ]; then
 fi
 
 if [ "$BUILD" == "install" ]; then
-    make -j$JOBS Release
+    gmake -j$JOBS Release
     exit_code=$?
     if [ $exit_code != 0 ]; then
       echo "there has been a problem compiling Release OF library"
diff --git a/scripts/linux/compilePG.sh b/scripts/linux/compilePG.sh
index 348f78c44..975008de8 100755
--- a/scripts/linux/compilePG.sh
+++ b/scripts/linux/compilePG.sh
@@ -1,10 +1,11 @@
-#!/bin/bash
+#!/usr/local/bin/bash
+set -x
 
 export LC_ALL=C
 
 OF_ROOT=$(cd $(dirname $0)/../..; pwd -P)
 
-make Release -C ${OF_ROOT}/apps/projectGenerator/commandLine
+gmake Release -C ${OF_ROOT}/apps/projectGenerator/commandLine
 ret=$?
 if [ $ret -ne 0 ]; then
   echo "There has been a problem compiling the command line projectGenerator."
diff --git a/scripts/linux/testAllExamples.sh b/scripts/linux/testAllExamples.sh
index 0017ee931..3a2c317d9 100755
--- a/scripts/linux/testAllExamples.sh
+++ b/scripts/linux/testAllExamples.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/local/bin/bash
 
 export LC_ALL=C
 cd ../../examples
@@ -18,13 +18,13 @@ do
         echo "-----------------------------------------------------------------"
         echo "building " + $example
         cd $example
-        make Debug
+        gmake Debug
         ret=$?
         if [ $ret -ne 0 ]; then
           echo error compiling $example
           exit
         fi
-        make Release
+        gmake Release
         ret=$?
         if [ $ret -ne 0 ]; then
           echo error compiling $example
